src/encoded/static/components/viz/AreaChart.js

```diff
@@ -372,7 +374,25 @@ export class GroupByDropdown extends React.PureComponent {
     constructor(props){
         super(props);
         this.onGroupBySelect = _.throttle(this.onGroupBySelect.bind(this), 1000);
-        this.onDateRangeSelect = _.throttle(this.onDateRangeSelect.bind(this), 1000);
+        this.onDateRangeSelect = this.onDateRangeSelect.bind(this);
+        //used as workaround to fix input type="date" unwanted reset bug
+        this.state = {
+            'tempDateRangeFrom': '',
+            'tempDateRangeTo': ''
+        };
+    }
+
+    componentDidUpdate(pastProps, pastState){
+        const { currentDateRangeFrom, currentDateRangeTo } = this.props;
+        // if current date range from/to changed, then force the temp values get reset
+        if (pastProps.currentDateRangeFrom !== currentDateRangeFrom || pastProps.currentDateRangeTo !== currentDateRangeTo) {
+            setTimeout(() => {
+                this.setState({
+                    'tempDateRangeFrom': currentDateRangeFrom,
+                    'tempDateRangeTo': currentDateRangeTo
+                });
+            }, 750);
```

Comment:
Hi Utku, this PR looks good to me! Like Bianca mentioned might be good to add comments in some places but not a blocking issue.

I was wondering what the purpose of the setTimeout here is? Are you delaying the state update as a style decision, or is it to make the component properly re-render?

Reply:
@crfmc Calling setState within componentDidUpdate is not a good practice, as it can potentially lead to infinite rendering loops. I regret to say that setTimeout hack was the final resort.